src/borrow_tracker/tree_borrows/mod.rs

```diff
@@ -252,10 +253,12 @@ trait EvalContextPrivExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         };
         log_creation(this, Some((alloc_id, base_offset, parent_prov)))?;
 
-        let orig_tag = match parent_prov {
-            ProvenanceExtra::Wildcard => return interp_ok(place.ptr().provenance), // TODO: handle wildcard pointers
-            ProvenanceExtra::Concrete(tag) => tag,
+        let (orig_tag, provenance) = match parent_prov {
```

Comment:
So `orig_tag` is exactly the same as `parent_prov` except you converted it from `ProvenanceExtra` to `Option`...? What's the point of that?

Also, `provenance` is way to imprecise as a name. There's so many provenances floating around here, this isn't useful. I'd suggest `new_prov`.